PybindGPU/device_array.h

```diff
@@ -107,6 +108,29 @@ class DeviceArray {
             host_allocated = false;
             device_allocated = false;
         };
+        
+        DeviceArray(size_t host_addr, size_t device_addr, std::vector<ssize_t> & shape)
+        : m_shape{shape}, device_allocated(false) {
+            // total size
+            m_size = std::accumulate(
+                shape.begin(), shape.end(), 1,
+                std::multiplies<ssize_t>()
+            );
+            // define array strides, assuming c-order
+            m_ndim = shape.size();
+            m_strides = std::vector<ssize_t>(m_ndim);
+            ssize_t stride = sizeof(T);
+            for (int i = m_ndim - 1; i >= 0; i--) {
+                m_strides[i] = stride;
+                stride = stride * shape[i];
+            }
+            // Just point to the given addresses
+            host_ptr = reinterpret_cast<T *>(host_addr);
+            device_ptr = reinterpret_cast<T *>(device_addr);
+            // allocation status
+            host_allocated = true;
```

Comment:
@monarin I suggest setting `host_allocated = false` as this is an aliasing constructor. But setting `host_allocated=False` and `device_allocated=False` you are preventing the underlying pointers from being freed when the alias goes out of scope.